components/dynamic/DataList/index.js

```diff
@@ -49,29 +55,54 @@ const DataList = ({
     section,
   });
 
-  if (isLoading || isError) return null;
+  function renderData() {
+    const { offset, total } = data;
+    const numberOfPages = Math.ceil(total / limit) || 1;
 
-  const { offset, total } = data;
-  const numberOfPages = Math.ceil(total / limit) || 1;
+    // if our page is out of bounds, we must instead go to the last page with data
+    if (offset > total) {
+      router.push({
+        pathname,
+        query: { ...pathParams, page: numberOfPages },
+      });
+    }
 
-  // if our page is out of bounds, we must instead go to the last page with data
-  if (offset > total) {
-    router.push({
-      pathname,
-      query: { ...pathParams, page: numberOfPages },
-    });
+    if (total === 0 && isRelatedList) return null;
+    if (total === 0) return <div>{t(`search-no-results`)}</div>;
+    return children(data);
   }
 
-  if (total === 0 && isRelatedList) return null;
+  function renderPagination() {
+    const { offset, page, total } = data;
 
-  if (total === 0)
+    if (limit >= total || section === "glossaryTerms" || isRelatedList)
+      return null;
     return (
-      <Container>
-        <div>{t(`search-no-results`)}</div>
-      </Container>
+      <Pagination limit={limit} offset={offset} page={page} total={total} />
     );
+  }
 
-  return children(data);
+  if (isError) return null;
+
+  return (
+    <>
+      <Container width={width}>
+        {header && <Styled.Header>{header}</Styled.Header>}
+        {isLoading && <Loader speed="fast" isVisible />}
+        {!isLoading && data && renderData()}
```

Comment:
With all of the common elements refactored out of the components that use `DataList`, and into `DataList` itself, was able to also toggle children and Loader visibility here, instead of import a loader into every component where `DataList` is used.  I think there's actually even more opportunity for refactoring these dynamic List type components, as well as also the GridBlock components.  There seems to be quite a lot of commonality about the items in the List/Gird, such that some of the slight differences might be better controlled with props instead of discrete components for every use case.